src/Swagger/Extractor/CollectionPostExtractor.php

```diff
@@ -0,0 +1,160 @@
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Swagger\Model\SwaggerDocumentation;
+use ApiPlatform\Core\Swagger\Model\SwaggerMetadata;
+use ApiPlatform\Core\Swagger\Util\SwaggerDefinitions;
+use Symfony\Component\HttpFoundation\Request;
+
+class CollectionPostExtractor implements SwaggerExtractorInterface
+{
+    /**
+     * @var SwaggerExtractorInterface
+     */
+    private $decoratedExtractor;
+
+    /**
+     * @var ResourceMetadataFactoryInterface
+     */
+    private $resourceMetadataFactory;
+
+    /**
+     * @var SwaggerDefinitions
+     */
+    private $swaggerDefinitions;
+
+    /**
+     * RouteDocumentationExtractor constructor.
+     *
+     * @param ResourceMetadataFactoryInterface $resourceMetadataFactory
+     * @param SwaggerDefinitions               $swaggerDefinitions
+     * @param SwaggerExtractorInterface        $decoratedExtractor
+     */
+    public function __construct(
+        ResourceMetadataFactoryInterface $resourceMetadataFactory,
+        SwaggerDefinitions $swaggerDefinitions,
+        SwaggerExtractorInterface $decoratedExtractor = null
+    ) {
+        $this->decoratedExtractor = $decoratedExtractor;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->swaggerDefinitions = $swaggerDefinitions;
+    }
+
+    public function extract(SwaggerMetadata $swaggerMetadata): SwaggerDocumentation
+    {
+        if (null !== $this->decoratedExtractor) {
+            $documentation = $this->decoratedExtractor->extract($swaggerMetadata);
+        } else {
+            $documentation = new SwaggerDocumentation();
+        }
+
+        if ($swaggerMetadata->isCollection() && in_array(Request::METHOD_POST, $swaggerMetadata->getMethods())) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($swaggerMetadata->getResourceClass());
+
+            $this->extractSummary($resourceMetadata, $documentation);
+            $this->extractResponses($swaggerMetadata, $documentation, $resourceMetadata);
+            $this->extractParameters($swaggerMetadata, $documentation, $resourceMetadata);
+
+            if (empty($documentation->getConsumes())) {
+                $documentation->setConsumes($swaggerMetadata->getMimeTypes());
+            }
+        }
+
+        return $documentation;
+    }
+
+    /**
+     * @param $documentation
+     * @param $resourceMetadata
+     */
+    protected function extractSummary(ResourceMetadata $resourceMetadata, SwaggerDocumentation $documentation)
+    {
+        if (null === $documentation->getSummary()) {
+            $documentation->setSummary(sprintf('Creates a %s resource.', $resourceMetadata->getShortName()));
+        }
+    }
+
+    /**
+     * @param SwaggerMetadata $swaggerMetadata
+     * @param $documentation
+     * @param $resourceMetadata
+     */
+    protected function extractResponses(
```

Comment:
We inline method prototypes (and we never use a protected method without a good use case)